stix/idl/io/stx_map2fits.pro

```diff
@@ -0,0 +1,285 @@
+  endif else begin
+    map = in_map
+  endelse
+
+  ; *************************************************************
+
+   if ~valid_map(map) || is_blank(file) then begin
+    pr_syntax,'map2fits,map,file'
+    return
+   endif
+   np=n_elements(map)
+
+;-- form output file name
+
+   cd,curr=cdir
+   break_file,file,dsk,dir,dfile,dext
+   if trim(dsk+dir) eq '' then fdir=cdir else fdir=dsk+dir
+   if ~file_test(fdir,/write) then begin
+    err='No write access to - '+fdir
+    message,err,/info
+    return
+   endif
+   if trim(dext) eq '' then fext='.fits' else fext=dext
+   filename=concat_dir(fdir,dfile)+fext
+   filename=filename.Replace("'","")
+   
+   if verbose then message,'Writing map to - '+filename,/info
+   use_rtime=tag_exist(map,'rtime')
+   in_tags=tag_names(map)
+   def_map,rmap
+   def_fits=['SIMPLE','BITPIX','NAXIS','NAXIS1','NAXIS2','DATE','FILENAME','END',$
+             'COMMENT','HISTORY']
+   ignore=[tag_names(rmap),def_fits]
+   delvarx,header
+   for i=0,np-1 do begin
+
+;-- unpack data
+
+    unpack_map,map[i],data,xp,yp,dx=cdelt1,dy=cdelt2,xc=xcen,yc=ycen,$
+      nx=naxis1,ny=naxis2
+
+;-- scale data?
+
+    if keyword_set(byte_scale) then bscale,data,top=255
+
+;-- add header for the output array.
+
+    fxhmake,header,data,/date
+
+;-- add FITS parameters CRPIX, CRVAL, etc.
+
+    crpix1=comp_fits_crpix(xcen,cdelt1,naxis1,0)
+    crpix2=comp_fits_crpix(ycen,cdelt2,naxis2,0)
+
+    fxaddpar, header, 'ctype1', 'solar_x','Solar X (cartesian west) axis'
+    fxaddpar, header, 'ctype2', 'solar_y','Solar Y (cartesian north) axis'
+
+;    fxaddpar, header, 'cunit1', 'arcsecs','Arcseconds from center of Sun'
+;    fxaddpar, header, 'cunit2', 'arcsecs','Arcseconds from center of Sun'
+
+    fxaddpar, header, 'crpix1', crpix1, 'Reference pixel along X dimension'
+    fxaddpar, header, 'crpix2', crpix2, 'Reference pixel along Y dimension'
+
+    fxaddpar, header, 'crval1',0, 'Reference position along X dimension'
+    fxaddpar, header, 'crval2',0, 'Reference position along Y dimension'
+
+    fxaddpar, header, 'cdelt1',cdelt1,'Increments along X dimension'
+    fxaddpar, header, 'cdelt2',cdelt2,'Increments along Y dimension'
+
+    if use_rtime then obs_time=map[i].rtime else obs_time=map[i].time
+
+    fxaddpar,header,'date_obs',obs_time,'Observation date'
+    fxaddpar,header,'date-obs',obs_time,'Observation date'
+    if tag_exist(map,'dur') then fxaddpar,header,'exptime',map[i].dur,'Exposure duration'
+    fxaddpar,header,'origin',map[i].id,'Data description'
+    
+    
+    ; ******************** ADAPTED FOR STIX ************************
+    
+    ; Extract the time and energy ranges from the map
+    time_range = [anytim(map.time)-map.dur/2.,anytim(map.time)+map.dur/2.]
+    ;energy_axis = map.energy_range
+    
+    ; Algorithm used
+    ;algo_used = map.image_alg
+    
+    ; If xy_shift is not provided, then put the average shift values in the header
+    if not keyword_set(xy_shift) then xy_shift = [26.1, 58.2]
```

Comment:
What are these used for all the position information should be in the WCS keywords

Reply:
Same question from me

Reply:
Thanks for reporting this. This is old stuff, when the aux FITS files were not yet ready. Now this has to be removed.
Additionally, the procedure needs the auxiliary data structure for more precise pointing information. This has to be done. I will do it for the next request